### PR TITLE
Datenbankbereinigung localized string

### PR DESCRIPTION
**Pull request by nittka**
WIP: I have started cleaning up variables either using the all-languages tag or turning variables to global ones. The first commit contains changes in the user databases. I have not dealt with all cases because I need feedback for the general pattern
* ads - very often the names are identical in all languages (and we don't have variables, yet; should ad names be "global" in general or rather get the same value in each language?
* programme and script template (original) titles without variables - if they are identical in all languages, turn them into global titles or rather all-languages

Edit: The general pattern (I tried to follow)
* randomcity -> global
* worldtime -> global
* contains variable or role reference -> all (role localization)
* identical fixed values -> global

### CODE REVIEW

**Review by nittka — COMMENTED** [review]
Of course, one option is not to make the global-all-decision for all cases and just leave two/three identical language tags.

**Comment by GWRon** [comment]
I am not sure but I consider the terms "global" and "all" and .. not distinguishable enough. It should be somehow made easier to understand the effect of using "all" or leaving out any language tag at all.

Leaving out is using "english" as it is the fallback and this means any subsequent use of variables will only check "english" of their subdefinitions.
Using "all" will resolve for de,en,pl (in addition to custom languages used in title/description itself).


So under this "understanding assumptions" using "english" is `<title>global title used in all languages</title` right? And as soon as I either have differences in the raw text already, or expressions/variables are used which can differ amongst languages, I would use `<title><all>${somevar} ${someothervar}</all></title>`.

Right?

Am not sure if we can then create a rule because any static text then by default can change from "all" to a global during loading (nothing changing/localizing content afterwards).

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 1]
```diff
@@ -8705,14 +8675,8 @@ The project, which was cancelled by a major producer, is now free to be realized
 			</description>
 			<children>
 				<scripttemplate index="0" product="2" licence_type="2" guid="7d13827e-63c7-4d87-a1e0-73d8a19b16f5">
-					<title>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:8708]: example for controversial case - global/all/leave as is

**Comment by nittka** [comment]
I guess there is very little debate about the case where variables are involved. Don't use "default" (or "fixed"? - better term than "global"?) except for a handful of exceptions - use of worldtime/randomcity/persongenerator expression as here an identical result for all languages is intended.

The reason why I am wondering about the "no variables" case is that possibly for the current translations everything is the same, so you could go without language tag. But once a new translation wanted to use a different text, you'd have to change the database entry from `<title>fixed</title>` to `<title><all>fixed</all><newLanguage...</title>` rather than just adding a language tag.
Also going from two identical language tags to <all> somewhat implies that it was already checked that the Polish version indeed is identical. So leaving those cases untouched might be better.

**Comment by GWRon** [comment]
yes, you need to change from 
```
<title>fixed</title>
```
to
```
<title>
  <all>the same for all default languages</all>
  <newLanguage>but differing here</newLanguage>
</title>
```
I am not sure if we could do differently at all -- without removing the `<var>${.blaFunc}</var>` support.

This aside "all" for me implies "applies to all languages" or "same content for all languages" which someone also can interpret as "global" (especially if you do not know about the difference - one entry for all - vs. - auto generation of entries for all languages).

Most "clear" is imo by explicitly defining entries for all used languages. And if you do not do that it means "no distinguish between languages" (the "global") which also means, the content is something not to "localize" (eg it stores a integer number or so). Which brings up the discussion about ... what is to localize and what not? title, description etc are most probably always to localize. Exception is if it contains an expression `${...}` because this makes the value non static in that moment and possibly "localized". Any "static text" could be "global" - yet it adds confusion if some language is to add which requires adjustments (for whatever reason). So for title/description maybe simple define entries for each language even if they are the same. At the end a tool could check for a given "language locale key" if there are entries missing a localized title or description... should not be something the db author is handling manually.

Maybe for variables it should be handled differently. Variables could benefit from the `<var>static non localized content</var>`. The author of the entry will most probably know what can be localized or not (indices, numeric percentages/amount of money, ...). If for some rare reason a language is added by someone else which prefers to have "hex values" then it is up to that author to change the `<var>` to a completely explicit `<var><de>...</de><en>...</en></var>` and from then on any other author will be aware that there are exceptions there.

Instead of this, we could use also a "default" instead of "all":
```
<var>
  <default>254|255</default>
  <leetlanguage>fe|ff</leetlanguage>
</var>
```
instead of saying "all" it becomes more clear that any "unspecified language" uses the default value. But this does not work with out "de,en,pl"-fixation (it works but has flaws "interpretation wise") and leads back to the discussion about "how to identify default languages" - not easy to simply "accept" that "default/all" just means "the ones someone fixed in base.database.bmx" (albeit we could add the language keys to the xml tvt-header...)

**Comment by nittka** [comment]
OK. Except for the main programmes file, I have gone through the files. Ads I have mostly left with a single language rather than using "all".
Programmes will take a while (about 800 findings out of 1400).

**Comment by nittka** [comment]
I propose to stop at this point. The programme definitions do not suffer from the variable problem (title with one language tag containing variables). That is, cleaning up by combining entries into one all-languages tag will not change anything for the game. We can leave that to later, if we find it important engough. The same is true for ads with only one language tag.

**Comment by scr0llbaer9000** [comment]
> randomcity -> global

as mentioned elsewhere, city names *could* be different... *theoretically*... like Munich vs München; not that we have examples with the city generator. But it's imaginable it could one day return different endings that would be typically different depending on the language, like -ow vs -ov (even when playing on the same map).

**Comment by nittka** [comment]
> city names could be different... theoretically...
Yes, maybe in a couple of years - requiring the map configuration and the expression evaluation to be refactored just for that use case. For now it would be nice if the generated titles (and usage in the database) were consistent.

**Comment by scr0llbaer9000** [comment]
can be merged from my side so that we can go forward.

**Pull request merged**